### PR TITLE
lib/runtime: support runtime v0.8

### LINKED ISSUE #1104: Update Runtime API for v0.8

### PR DESCRIPTION
**Pull request by noot**
## Changes

- update runtime package to have interfaces and functionality for both `Instance` (v0.8) and `LegacyInstance` (v0.6)
- stub imports for v0.8 runtime, implement some needed by API calls
- test `Core_version` call for v0.8, currently working only with wasmer
- node still runs legacy runtime by default, will update in a separate PR

- TODO: test all exported funcs for v0.8

## Tests

```
go test ./lib/runtime/...
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- related to #1104

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]
Great work, lgtm.

File: lib/runtime/wasmtime/instance.go [hunk 1]
```diff
@@ -148,6 +212,7 @@ func (in *Instance) exec(function string, data []byte) ([]byte, error) {
 	length := int32(ret >> 32)
 	offset := int32(ret)
 
-	runtime.KeepAlive(mem)
+	runtime.KeepAlive(in.mem)
+	fmt.Println(memdata[:2048])
```
Comment (arijitAD) [inline lib/runtime/wasmtime/instance.go:216]: Would this print something meaningful?

File: lib/runtime/wasmtime/instance.go [hunk 2]
```diff
@@ -130,8 +194,8 @@ func (in *Instance) exec(function string, data []byte) ([]byte, error) {
 	}
 	defer ctx.Allocator.Clear()
 
-	mem := in.vm.GetExport("memory").Memory()
-	memdata := mem.UnsafeData()
+	//mem := in.vm.GetExport("memory").Memory()
```
Comment (arijitAD) [inline lib/runtime/wasmtime/instance.go:197]: Is this left commented intentionally?

**Review by arijitAD — APPROVED** [review]

File: lib/runtime/wasmtime/instance.go (hunk 1 above)
Comment (noot) [reply lib/runtime/wasmtime/instance.go:216]: no, will remove!

File: lib/runtime/wasmtime/instance.go (hunk 2 above)
Comment (noot) [reply lib/runtime/wasmtime/instance.go:197]: no, will remove!

**Pull request merged**